domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt

```diff
@@ -156,16 +161,18 @@ class TopicController @Inject constructor(
     topicId: String,
     storyId: String,
     explorationId: String
-  ): DataProvider<ChapterSummary> {
-    return dataProviders.createInMemoryDataProviderAsync(GET_CHAPTER_PROVIDER_ID) {
-      return@createInMemoryDataProviderAsync AsyncResult.success(
-        fetchChapter(
-          topicId,
-          storyId,
-          explorationId
+  ): LiveData<AsyncResult<ChapterSummary>> {
+    return MutableLiveData(
+      try {
+        AsyncResult.success(fetchChapter(topicId, storyId, explorationId))
+      } catch (e: Exception) {
```

Comment:
We shouldn't be catching a generic exception like this--how do you know the correct one was caught? Also, it's considered an antipattern to use exceptions for control flow. Instead, return a result or introduce a mechanism to check for existence to make the determination that the chapter doesn't exist.

Reply:
updated retrieveChapter

Reply:
updated retrieveChapter